### PR TITLE
Added: '-verbose' and '-vverbose' flags

### LINKED ISSUE #17: Change the output/stdout when running terracognita also adding VERBOSE mode
Right now the output it's basically the logs from `terrafrom`, we should show those when we add the `--verbose` flag, which does not exists ATM.

The desired output would be, the one without the `--verbose`, this one:

```
Jun 19 12:17:01 INFO: Importing security_group [35/35] Done!
Jun 19 12:17:01 INFO: Importing aws_instance [10/35] ...
```

We still have some things TBD:

* Do we want the TF logs on `--verbose`? Or we want something custom?
* The TF logs are important? If so where/when do we want them?

**EDIT:**

We'll have 2 flags `-v` and `-vv`:

* Without any flag we'll have something like this output
```
Jun 19 12:17:01 INFO: Importing security_group [35/35] Done!
Jun 19 12:17:01 INFO: Importing aws_instance [10/35] ...
```
* With `-v` we'll have the logs we had before with more custom logs (Debug) like this

```
Jun 19 12:17:01 DEBUG: Starting terracognita for region R - resources: X, Y, Z
Jun 19 12:17:01 INFO: Importing security_group [35/35] Done!
...
Jun 19 12:17:01 INFO: Importing aws_instance [9/35] ...
Jun 19 12:17:01 DEBUG: Importing aws_instance i-123lasdjk2
Jun 19 12:17:01 DEBUG: Writing HCL aws_instance i-123lasdjk2
Jun 19 12:17:01 DEBUG: Writing State aws_instance i-123lasdjk2
Jun 19 12:17:01 DEBUG: Finished importing aws_instance i-123lasdjk2
Jun 19 12:17:01 INFO: Importing aws_instance [10/35] ...
```

* With `-d` we'll have the logs we had before with the TF ones:

```
2019/06/20 12:13:06 [INFO] Building AWS auth structure
2019/06/20 12:13:06 [INFO] Setting AWS metadata API timeout to 100ms
2019/06/20 12:13:07 [INFO] Ignoring AWS metadata API endpoint at default location as it doesn't return any instance-id
2019/06/20 12:13:07 [INFO] AWS Auth provider used: "StaticProvider"
2019/06/20 12:13:07 [DEBUG] Trying to get account information via sts:GetCallerIdentity
2019/06/20 12:13:07 [DEBUG] Trying to get account information via sts:GetCallerIdentity
Jun 19 12:17:01 DEBUG: Starting terracognita for region R - resources: X, Y, Z
Jun 19 12:17:01 INFO: Importing security_group [35/35] Done!
...
Jun 19 12:17:01 INFO: Importing aws_instance [9/35] ...
Jun 19 12:17:01 DEBUG: Importing aws_instance i-123lasdjk2
Jun 19 12:17:01 DEBUG: Writing HCL aws_instance i-123lasdjk2
Jun 19 12:17:01 DEBUG: Writing State aws_instance i-123lasdjk2
Jun 19 12:17:01 DEBUG: Finished importing aws_instance i-123lasdjk2
Jun 19 12:17:01 INFO: Importing aws_instance [10/35] ...
```

### PR DESCRIPTION
**Pull request by xescugc**
The `-verbose` uses go-kit/log to log information of what's currently happening on the import.

The `-debug` adds the `TF_LOG` so it also logs TF logs.

Closes #17

### CODE REVIEW

File: cmd/root.go [hunk 1]
```diff
@@ -15,12 +17,26 @@ var (
...
+	logsOut          io.Writer
 
 	// RootCmd it's the entry command for the cmd on terracognita
 	RootCmd = &cobra.Command{
 		Use:   "terracognita",
 		Short: "Reads from Providers and generates a Terraform configuration",
 		Long:  "Reads from Providers and generates a Terraform configuration, all the flags can be used also with ENV (ex: --access-key == ACCESS_KEY)",
+		PersistentPreRun: func(cmd *cobra.Command, args []string) {
+			// Initialize the logs by setting by default the logs
+			// to Stdout, but if 'v' or 'd' it's defined the logger
```
Comment (xlr-8) [inline cmd/root.go]: is defined*

File: filter/filter.go [hunk 2]
```diff
@@ -27,6 +31,15 @@ func (f *Filter) IsExcluded(v string) bool {
 	return ok
 }
 
+// String returns an stringification of the Filter
```
Comment (xlr-8) [inline filter/filter.go]: Could be nice to have it in `-v` & `-d` output,  I haven't when running the code.

File: log/log.go [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+	"github.com/hashicorp/terraform/helper/logging"
+)
+
+var logger kitlog.Logger
+var once sync.Once
+
+// Init initializes the log, it can only be called once,
+// repetitive calls to it will not change it.
+// It also set the level of vebosity of the
+// Teffaform logs via tflogs, if true it'll
```
Comment (xlr-8) [inline log/log.go]: Terraform*

File: cmd/root.go [hunk 4]
```diff
@@ -75,6 +92,12 @@ func init() {
 
 	RootCmd.PersistentFlags().StringSliceVarP(&exclude, "exclude", "e", []string{}, "List of resources to not import, this names are the ones on TF (ex: aws_instance). If not set then means that none the resources will be excluded")
 	_ = viper.BindPFlag("exclude", RootCmd.PersistentFlags().Lookup("exclude"))
+
+	RootCmd.PersistentFlags().BoolP("verbose", "v", false, "Activate the verbose mode")
+	_ = viper.BindPFlag("verbose", RootCmd.PersistentFlags().Lookup("verbose"))
+
+	RootCmd.PersistentFlags().BoolP("debug", "d", false, "Activate the debug mode wich includes TF logs via TF_LOG configuration https://www.terraform.io/docs/internals/debugging.html")
```
Comment (xlr-8) [inline cmd/root.go]: I'd update the message to be more explicit about the content of TF_LOG, which could be missleading for beginner. 
I'd put something like: `TF_LOG=TRACE|DEBUG|INFO|WARN|ERROR` which is more explicit, while still keeping the link to terraform.io.

File: provider/import.go [hunk 5]
```diff
@@ -52,33 +84,51 @@ func Import(ctx context.Context, p Provider, hcl, tfstate writer.Writer, f *filt
...
 
 			if tfstate != nil {
+				logger.Log("msg", "calculating TFState")
 				err = r.State(tfstate)
 				if err != nil {
 					return errors.Wrapf(err, "error while calculating the Satate of resource %q", t)
 				}
 			}
 		}
+		if resourceLen != 0 {
```
Comment (xlr-8) [inline provider/import.go]: I'd even say `> 0`, it's not likely to happen whatsoever but it filters any negative entity.

File: filter/filter.go (hunk 2 above)
Comment (xescugc) [reply filter/filter.go]: You do: `ts=2019-07-18T09:30:15.38006452Z caller=import.go:47 func=provider.Import filters="\n\tTags:    [],\n\tInclude: [aws_iam_user],\n\tExclude: [],\n"`

**Comment by xescugc** [comment]
Done

**Pull request merged**